src/common/TableOfContents.ts

```diff
@@ -0,0 +1,12 @@
+export interface HeadingLink {
+  content: string
+  anchorLink: string
+  level: number
+}
+
+export interface TableOfContentsProps {
+  type: "tableofcontents"
```

Comment:
why dont we use camelcase for naming convention?

Reply:
sure, @dcshzj was there any reason why you went with this naming convention originally? If not I can change them all in another PR

Reply:
No preference here, originally went for lowercase alphabet cos I saw it more like an ID. Can change if needed.